pkg/provider/ip/provider.go

```diff
@@ -111,6 +111,20 @@ func (p *ipv4Provider) InitResource(instance ec2.EC2Instance) error {
 
 	podToResourceMap := map[string]pool.Resource{}
 	usedIPSet := map[string]struct{}{}
+
+	// Construct map of all possible IPs to prefix for each assigned prefix
+	ipToPrefixMap := make(map[string]string)
+	usedPrefixSet := make(map[string]struct{})
+	for _, prefix := range presentPrefixes {
+		prefixIPs, err := utils.DeconstructIPsFromPrefix(prefix)
+		if err != nil {
+			continue
```

Comment:
Should we at least log the error?

Reply:
Thanks for the suggestion. I've added the log here.